### ISSUE TITLE
Create separate lib for Security ("Auth"?) module.

### ISSUE THREAD

**Issue by jessekelly881** [sukovanej/effect-http#547]
The Security lib might work better as a separate (transport layer independent) library that works not only with with effect-http but also with pure @effect/platform http, rpc, and other transport layers. That's how I would like to use it anyway. :)

**Comment by sukovanej** [comment]
Yeah, I already thought about it. That holds for the middlewares like CORS in the codebase as well. Now that I see an outside interest in that, I might actually do it 😄.

**Comment by sukovanej** [comment]
Hey, done within the latest release. Readme, examples and probably some combinators enabling easy integration with a /platform are missing tho.

**Comment by jessekelly881** [comment]
Looks fantastic! Thanks @sukovanej. Let me know if you need a hand with it. I'd be happy to help. :)

**Comment by sukovanej** [comment]
We discussed potential merge of this lib to the official effect monorepo. Independently of how that'll turn out, we talked with fubhy about the separation of packages and the result is we should rather aim for a simple setup with a single platform-independent package a few platform-specific ones. So I will very probably merge the `effect-http-error` and `effect-http-security` back. `effect-http` is practically dependency-free, there is `@effect/platform`, `@effect/schema` and `effect` as peer deps so it doesn't really make any difference.
